cmd/stackdriver-prometheus-sidecar/main.go

```diff
@@ -206,7 +206,6 @@ func main() {
 	a.Flag("config-file", "A configuration file.").StringVar(&cfg.ConfigFilename)
 
 	projectID := a.Flag("stackdriver.project-id", "The Google project ID where Stackdriver will store the metrics.").
-		Required().
```

Comment:
As I recall, making the project id a required flag was an intentional choice to make sure there are no surprises in non-GCE environments.
Even if the flag is required, isn't `""` a valid value? So explicitly supplying an empty value via `--stackdriver.project-id=` on the command line should cause the sidecar to query the metadata server on GCE, right?
@StevenYCChou might want to chime in here.

Reply:
Oh, of course! Sorry, you're totally right. I missed the fact that `""` is valid value. Everything is right then.

Closing the PR, sorry for noise.